hector_pose_prediction_ros/include/hector_pose_prediction_ros/visualization.h

```diff
@@ -0,0 +1,226 @@
+#ifndef HECTOR_POSE_PREDICTION_ROS_VISUALIZATION_H
+#define HECTOR_POSE_PREDICTION_ROS_VISUALIZATION_H
+
+#include <eigen_conversions/eigen_msg.h>
+#include <hector_math/types.h>
+#include <hector_pose_prediction_interface/types.h>
+#include <std_msgs/ColorRGBA.h>
+#include <visualization_msgs/MarkerArray.h>
+
+namespace hector_pose_prediction_ros
+{
+namespace visualization
+{
+template<typename Scalar>
+void drawSupportPolygonImpl( visualization_msgs::MarkerArray &marker_array,
+                             const hector_math::Vector3List<Scalar> &contact_hull_points,
+                             const std::vector<std_msgs::ColorRGBA> &colors,
+                             const std::string &frame_id, const std::string &ns )
+{
+  marker_array.markers.reserve( marker_array.markers.size() + contact_hull_points.size() );
+  for ( size_t i = 0; i < contact_hull_points.size(); i++ ) {
+    visualization_msgs::Marker marker;
+    marker.type = visualization_msgs::Marker::ARROW;
+    marker.action = visualization_msgs::Marker::ADD;
+    marker.scale.x = 0.02;
+    marker.scale.y = 0.00001;
+    marker.scale.z = 0.00001;
+
+    marker.color = colors[i];
+    marker.header.frame_id = frame_id;
+    marker.ns = ns;
+    marker.id = static_cast<int32_t>( i );
+
+    marker.pose.orientation.w = 1.0;
+
+    geometry_msgs::Point point_start;
+    tf::pointEigenToMsg( contact_hull_points[i].template cast<double>(), point_start );
+    marker.points.push_back( point_start );
+    int ip1 = ( i + 1 ) % contact_hull_points.size(); // wrap around
+    geometry_msgs::Point point_end;
+    tf::pointEigenToMsg( contact_hull_points[ip1].template cast<double>(), point_end );
+    marker.points.push_back( point_end );
+    marker_array.markers.push_back( marker );
+  }
+}
+
+template<typename Scalar>
+void drawSupportPolygon( visualization_msgs::MarkerArray &marker_array,
+                         const hector_math::Vector3List<Scalar> &contact_hull_points,
+                         Eigen::Vector4f color, const std::string &frame_id, const std::string &ns )
+{
+  std_msgs::ColorRGBA color_msg;
+  color_msg.r = color( 0 );
+  color_msg.g = color( 1 );
+  color_msg.b = color( 2 );
+  color_msg.a = color( 3 );
+  drawSupportPolygonImpl( marker_array, contact_hull_points,
+                          std::vector<std_msgs::ColorRGBA>( contact_hull_points.size(), color_msg ),
+                          frame_id, ns );
+}
+
+template<typename Scalar>
+void drawSupportPolygonWithStability( visualization_msgs::MarkerArray &marker_array,
+                                      const hector_math::Vector3List<Scalar> &contact_hull_points,
+                                      const std::vector<Scalar> &edge_stabilities,
+                                      Scalar max_stability, const std::string &frame_id,
+                                      const std::string &ns )
+{
+  if ( contact_hull_points.size() != edge_stabilities.size() ) {
+    return;
```

Comment:
You should throw if the function is used incorrectly and won't work as intended.